javascript/javascript2/week1/Js2Week1.js

```diff
@@ -0,0 +1,102 @@
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+const lengthArr = [];
+let min;
+let shortestWord;
+
+function findShortestWord() {
+
+    for (let i = 0; i < danishWords.length; i++) {
+        lengthItem = danishWords[i].length;
+        lengthArr.push(lengthItem);
+        lengthArr.sort((a, b) => a - b)
+        min = lengthArr[0];
+        if (danishWords[i].length == min) {
+
+            shortestWord = danishWords[i];
+        }
+
+    }
+    return shortestWord;
+}
+
+shortestWord = findShortestWord();
+console.log(shortestWord);
+
+
+
+
+/*Find and count the Danish letters
+Find the individual number and the total number of Danish letters in a string.
+
+const danishString = "Jeg har en blå bil";
+notThisFunctionName(danishString); // returns {total: 1, å: 1}
+
+const danishString2 = "Blå grød med røde bær";
+notThisFunctionName(danishString2); // returns {total: 4, æ: 1, ø: 2, å: 1}*/
+
+const danishLetters = ['å', 'ø', 'æ',]
+const danishString = "Jeg har en blå bil";
+const danishString2 = "Blå grød med røde bær";
+
+function findDanishLetters(string) {
+    const resultObj = {};
+    let sum = 0;
+    string = string.split('');
+    console.log(string);
+    for (i = 0; i < string.length; i++) {
+        if (danishLetters.includes(string[i])) {
+
+            resultObj[string[i]];
```

Comment:
This does not seem necessary

Reply:
I am confused here. I cant understand how it works (as it definitely works without it, and it also seems unnecessary for you , so for sure there is some logic for it not to be there). But I am finding it difficult to understand how the empty object then includes the danish letters as the (key). Only if the put the above, will the following statement
 if (danishLetterArray[i] in resultObj)
will be true right, then how does it work. I hope I am able to make you understand what my issue is.

Reply:
The empty object doesnt have the danish letter key to begin with. But if it is not there, you set it on line 56. If it is there, you increment its value by 1 on line 53. The expression `resultObj[string[i]];` will reference a key which wont be there to begin with, and it does nothing with its value. If i didnt answer your question right, ask again :)

Reply:
Ohhhh yes. Now I get it, as line 56 does the part. That was so silly of me that I have been struggling to understand this. Thanks a lot for your explanation